src/ng/location.js

```diff
@@ -143,18 +151,8 @@ function LocationHtml5Url(appBase, appBaseNoFile, basePrefix) {
     this.$$compose();
   };
 
-  /**
-   * Compose url and update `absUrl` property
-   * @private
-   */
-  this.$$compose = function() {
-    var search = toKeyValue(this.$$search),
-        hash = this.$$hash ? '#' + encodeUriSegment(this.$$hash) : '';
-
-    this.$$url = encodePath(this.$$path) + (search ? '?' + search : '') + hash;
-    this.$$absUrl = appBaseNoFile + this.$$url.substr(1); // first char is always '/'
-
-    this.$$urlUpdatedByLocation = true;
+  this.$$normalizeUrl = function(path) {
```

Comment:
I think `$$normalizeUrl`'s argument should be named `url` (not `path`), because that is what it is (in `$location` terms).

Reply:
Hmm... even though it's actually (part of) a path? The main thing this method does is append a base path to the argument. Seems like this is more of a path2url method...

Reply:
It is not necessarilt part of the path. It also contains query and hash fragment. So, technically it is part the browser URL.
But in `$location` terms, it is the URL (which is why `$$normalizeUrl()` is called with `$location.$$url` :grin:).